.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -7,7 +7,9 @@ behind them.
 
   - [ ] I've read [CONTRIBUTING.md](https://github.com/xmonad/xmonad/blob/master/CONTRIBUTING.md)
 
-  - [ ] I tested my changes with [xmonad-testing](https://github.com/xmonad/xmonad-testing)
+  - I tested my changes
+    - [ ] With `xmonad` version XXX (commit XXX if using git)
+    - [ ] With `xmonad-contrib` commit XXX
```

Comment:
What's the point of asking for xmonad-contrib commit in a pull request? All the info is available in the git history itself. This whole `I tested my changes with xmonad/xmonad-contrib version XXX` thing makes sense only in the issue template (where it's sorely missing), but not here, IMO.

(xmonad version would possibly make sense in the PR template, if xmonad was actually changing, but it isn't really, and people who make PRs that actually depend on some recently merged or work in progress functionality of xmonad core or X11 usually mention it anyway)

Reply:
> What's the point of asking for xmonad-contrib commit in a pull request? All the info is available in the git history itself. 

I saw it as more of a bookkeeping device for longer PRs.  This way, we at least know a version where it definitely worked, should any regressions be introduced later.  Maybe that's overthinking it...

> (xmonad version would possibly make sense in the PR template, if xmonad was actually changing, but it isn't really, and people who make PRs that actually depend on some recently merged or work in progress functionality of xmonad core or X11 usually mention it anyway)

You're probably right on that one. Xmonad doesn't change too much and so perhaps it's superfluous to ask a specific commit there.

> This whole `I tested my changes with xmonad/xmonad-contrib version XXX` thing makes sense only in the issue template (where it's sorely missing), but not here, IMO.

That's a good idea!

Reply:
@liskin so what do you think?  Keep it like this?  Completely remove the "I've tested my changes" from the pull request?  Have a passive-aggressive "I've tested my changes at all" there? :)

Reply:
Unless we have an alternative that is better, I'd say drop it.

We could possibly try encouraging people to add property/unit tests where that makes sense, but then we risk "incidents" like #407/#430 stalling for half a year when the submitter (in this case me) procrastinates on adding the tests. :-)

Or we could try to improve xmonad-testing into something that is actually useful. Having contrib modules be tested somehow, even if that's just testing that configuration samples typecheck, would be nice, but right now it's more hassle than it's worth. Ideally this would be much less pain, like a doctest or something that tests the docs in xmonad-contrib effortlessly. But all this requires hard work that someone would have to put in, and I don't think I have the time for that, so right now the best way forward is to either drop the line from the PR template, or replace with something like "I've considered how to best test this (property, unit, manually, …) and concluded that …" to let people know that some testing is encouraged and that they should at least think about it for a while.

Reply:
> or replace with something like "I've considered how to best test this (property, unit, manually, …) and concluded that …" to let people know that some testing is encouraged and that they should at least think about it for a while.

Oh I really like that, thanks!